dandi/bids_validator_deno/__init__.py

```diff
@@ -0,0 +1,445 @@
+    if result.returncode not in range(0, 2) or result.stderr != "":
+        raise ValidatorError(
+            result.args,
+            result.returncode,
+            result.stdout,
+            result.stderr,
+            outfile_content,
+        )
+
+    assert (
+        outfile_content is not None
+    ), "`outfile_content` should not be None when validation is successful"
+
+    # Parse the content, in JSON format, of the outfile
+    return BidsValidationResult.model_validate_json(outfile_content, strict=True)
+
+
+def get_version() -> str:
+    """
+    Return the version of the deno-compiled BIDS validator
+
+    Returns
+    -------
+    str
+        The version of the deno-compiled BIDS validator
+    """
+    return version(DISTRIBUTION_NAME)
+
+
+def _harmonize(
+    bv_result: BidsValidationResult, ds_path: Path
+) -> list[ValidationResult]:
+    """
+    Harmonize a `BidsValidationResult` object into a list of `ValidationResult` objects
+
+    Parameters
+    ----------
+    bv_result : BidsValidationResult
+        The `BidsValidationResult` object representing the result of the validation
+        using the deno-compiled BIDS validator
+    ds_path : Path
+        The path to the dataset that has been validated to produce the `bv_result`
+        object
+
+    Returns
+    -------
+    list[ValidationResult]
+        A list of `ValidationResult` objects in which each object represents an issue
+        in the validation result.
+    """
+    # Ensure the path is absolute and in canonical form
+    ds_path = ds_path.resolve()
+
+    issues = bv_result.issues.issues
+    code_messages = bv_result.issues.codeMessages
+    schema_version = bv_result.summary.schemaVersion
+    dandiset_path = find_parent_directory_containing("dandiset.yaml", ds_path)
+
+    origin = Origin(
+        type=OriginType.VALIDATION,
+        validator=Validator.bids_validator_deno,
+        validator_version=get_version(),
+        standard=Standard.BIDS,
+        # todo: the BIDSVersion is unavailable through the validator;
+        #   However, the schema version is actually more precise
+        standard_version=schema_version,
+    )
+
+    results: list[ValidationResult] = []
+    for issue in issues:
+        severity = issue.severity
+        if severity is BidsSeverity.ignore:
+            # Ignore issues with severity "ignore"
+            # TODO: If we want to include these issues, we will have to add a new value
+            #   to the Severity enum.
+            continue
+
+        # The absolute path to the file or directory that the issue is related to
+        issue_path = _get_path(issue, ds_path)
+
+        results.append(
+            ValidationResult(
+                id=f"BIDS.{issue.code}",
+                origin=origin,
+                scope=_get_scope(issue_path),
+                origin_result=issue,  # TODO: it may be more useful if set to `bv_result`
```

Comment:
nothing TODO here probably, hence:

```suggestion
                origin_result=issue,  # we store only the issue, not entire bv_result with more context
```

Reply:
Suggesting taken with some modification.